var/spack/repos/builtin/packages/foam-extend/package.py

```diff
@@ -40,207 +130,234 @@ class FoamExtend(Package):
     version('3.1', git='http://git.code.sf.net/p/foam-extend/foam-extend-3.1')
     version('3.0', git='http://git.code.sf.net/p/foam-extend/foam-extend-3.0')
 
-    variant('paraview', default=False, description='Enable ParaFOAM')
-    variant(
-        'scotch', default=True,
-        description='Activate Scotch as a possible decomposition library')
-    variant(
-        'ptscotch', default=True,
-        description='Activate PT-Scotch as a possible decomposition library')
-    variant(
-        'metis', default=True,
-        description='Activate Metis as a possible decomposition library')
-    variant(
-        'parmetis', default=True,
-        description='Activate Parmetis as a possible decomposition library')
-    variant(
-        'parmgridgen', default=True,
-        description='Activate Parmgridgen support')
-    variant(
-        'source', default=True,
-        description='Installs also the source folder')
+    # variant('int64', default=False,
+    #         description='Compile with 64-bit labels')
+    variant('float32', default=False,
+            description='Compile with 32-bit scalar (single-precision)')
+
+    variant('paraview', default=False,
+            description='Build paraview plugins (eg, paraFoam)')
+    variant('scotch', default=True,
+            description='With scotch for decomposition')
+    variant('ptscotch', default=True,
+            description='With ptscotch for decomposition')
+    variant('metis', default=True,
+            description='With metis for decomposition')
+    variant('parmetis', default=True,
+            description='With parmetis for decomposition')
+    variant('parmgridgen', default=True,
+            description='With parmgridgen support')
+    variant('source', default=True,
+            description='Install library/application sources and tutorials')
 
     supported_compilers = {'clang': 'Clang', 'gcc': 'Gcc', 'intel': 'Icc'}
 
+    provides('openfoam')
     depends_on('mpi')
     depends_on('python')
-    depends_on('flex')
     depends_on('zlib')
+    depends_on('flex@:2.6.1')  # <- restriction due to scotch
```

Comment:
@olesenm: Ok I think that is a bug -- the version is probably getting set too soon in the concretizer.  I have some forthcoming work to iron out issues like this, but your constraint should work for now.